src/git-source-provider.ts

```diff
@@ -245,6 +250,13 @@ export async function cleanup(repositoryPath: string): Promise<void> {
   // Remove auth
   const authHelper = gitAuthHelper.createAuthHelper(git)
   await authHelper.removeAuth()
+
+  // always do this last, otherwise other git commands may fail
+  await git
+    .tryConfigUnsetValue('safe.directory', `'${repositoryPath}$'`, true)
+    .catch(error => {
```

Comment:
will `tryConfigUnsetValue` ever throw? or we should check the return value from `tryConfigUnsetValue()`?